### PR TITLE
Move plugin to src

### PR DESCRIPTION
**Pull request by zeroadam**
Move include/plugin.php to Addon and Theme class

Related to #3878

### CODE REVIEW

**Comment by zeroadam** [comment]
This one actually works!!! Starting over was the right move

**Comment by annando** [comment]
You don't want to change table fields in the database, won't you? Please don't do that.

**Comment by zeroadam** [comment]
I made sure to stay away from the database

**Comment by MrPetovan** [comment]
> You don't want to change table fields in the database, won't you? Please don't do that.

Yes I do. If we change from `plugin` to `addon`, then I sure don't want us to keep that one occurrence in the DB "for historical reasons".

**Comment by annando** [comment]
Problem is that we then need some update function that moves the content from the one field into the other one.

**Comment by MrPetovan** [comment]
I still like my "plugin" => "addon" idea that I first emitted for renaming tables.

File: doc/Plugins.md [hunk 1]
```diff
@@ -69,39 +69,39 @@ Remember to declare it with '&' if you wish to alter it.
 Modules
 ---
 
-Plugins/addons may also act as "modules" and intercept all page requests for a given URL path.
-In order for a plugin to act as a module it needs to define a function "plugin_name_module()" which takes no arguments and needs not do anything.
+Addons/addons may also act as "modules" and intercept all page requests for a given URL path.
```
Comment (MrPetovan) [inline doc/Plugins.md]: Just "Addons".

File: doc/Plugins.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-Friendica Addon/Plugin development
+Friendica Addon development
```
Comment (MrPetovan) [inline doc/Plugins.md]: This file needs to be renamed `Addons.md` as well.

File: doc/Plugins.md [hunk 3]
```diff
@@ -7,32 +7,32 @@ Please see the sample addon 'randplace' for a working example of using some of t
...
 Modules work by intercepting specific page requests (by URL path).
 
-Plugin names cannot contain spaces or other punctuation and are used as filenames and function names.
+Addon names cannot contain spaces or other punctuation and are used as filenames and function names.
 You may supply a "friendly" name within the comment block.
-Each addon must contain both an install and an uninstall function based on the addon/plugin name.
-For instance "plugin1name_install()".
-These two functions take no arguments and are usually responsible for registering (and unregistering) event hooks that your plugin will require.
-The install and uninstall functions will also be called (i.e. re-installed) if the plugin changes after installation.
+Each addon must contain both an install and an uninstall function based on the addon/addon name.
```
Comment (MrPetovan) [inline doc/Plugins.md]: Just `addon`.

File: doc/database/db_addon.md [hunk 4]
```diff
@@ -4,14 +4,14 @@ Table addon
...
+| name          | addon base (file)name                        | char(255)  | NO   |     |         |                 |
 | version       | currently unused                              | char(255)  | NO   |     |         |                 |
 | installed     | currently always 1                            | tinyint(1) | NO   |     | 0       |                 |
 | hidden        | currently unused                              | tinyint(1) | NO   |     | 0       |                 |
 | timestamp     | file timestamp to check for reloads           | bigint(20) | NO   |     | 0       |                 |
 | plugin_admin  | 1 = has admin config, 0 = has no admin config | tinyint(1) | NO   |     | 0       |                 |
 
 Notes:
-These are addons which have been enabled by the site administrator on the admin/plugin page
+These are addons which have been enabled by the site administrator on the admin/addon page
```
Comment (MrPetovan) [inline doc/database/db_addon.md]: Just `addon`.

File: INSTALL.txt [hunk 5]
```diff
@@ -123,17 +123,17 @@ look like (changing it to the correct PHP location)
...
 two users and a handful of contacts. To do this, edit the file 
-".htconfig.php" and look for a line describing your plugins. On a fresh 
+".htconfig.php" and look for a line describing your addons. On a fresh 
 installation, it will look like
 
 $a->config['system']['addon'] = 'js_upload';
 
 This indicates the "js_upload" addon module is enabled. You may add additional 
-addons/plugins using this same line in the configuration file. Change it to 
+addons/addons using this same line in the configuration file. Change it to
```
Comment (MrPetovan) [inline INSTALL.txt]: Just `addons`.

**Comment by zeroadam** [comment]
Resolving conflicts is what got me into trouble last time. I know these are from recent changes you made in your last pr/merge. I can always go through these two files again afterwards.

**Comment by MrPetovan** [comment]
After what? I can't merge until you fix the conflicts.

**Comment by zeroadam** [comment]
Oooh, i see. I thought you could merge, "using mine" meaning ignoring my changes to those two files.

**Comment by zeroadam** [comment]
Wow!!! It's a miracle :-D

**Comment by MrPetovan** [comment]
Ha ha ha!

**Comment by MrPetovan** [comment]
How are you doing your rebases? All the deleted vendor files are present in the diff although they shouldn't.
Normal way:

````
$ git remote add friendica http://github.com/friendica/friendica
$ git fetch --all
$ git checkout Plugins-#3878
$ git rebase friendica/develop
````

What did you do exactly?

**Comment by zeroadam** [comment]
I am using gitkraken. I basically right click on upstream/develop and click rebase <mybranch> onto upstream/develop

**Comment by MrPetovan** [comment]
Isn't upstream/develop your own repository? Did you rebase it onto friendica/develop?

**Comment by zeroadam** [comment]
No, origin/develop is my repo. upstream is friendica/friendica.

**Comment by zeroadam** [comment]
You want me to reset --hard and rebase via command line?

**Comment by MrPetovan** [comment]
Can you please close this PR and try to open another one from the GitHub interface?

**Comment by zeroadam** [comment]
Closing for new PR

**Pull request closed without merging**